staging/src/k8s.io/apiserver/pkg/util/shufflesharding/shufflesharding_test.go

```diff
@@ -279,26 +282,43 @@ func BenchmarkShuffleAndDealToSlice(b *testing.B) {
 }
 
 func TestUniformDistribution(t *testing.T) {
-	deckSize, handSize := int32(128), int32(3)
-	handCoordinateMap := make(map[int]int)
+	var aHand [3]int
+	deckSize := int32(128)
+	handCoordinateMap := make(map[int]int) // maps coded hand to count of times seen
 
-	allCoordinateCount := 128 * 127 * 126 / 6
+	fallingFactorial := 128 * 127 * 126
+	allCoordinateCount := fallingFactorial / 6
+	hashMax := uint64(16 << uint32(math.Ceil(math.Log2(float64(fallingFactorial)))))
+	center := float64(hashMax) / float64(allCoordinateCount)
 
-	for i := 0; i < allCoordinateCount*16; i++ {
-		hands := ShuffleAndDealToSlice(rand.Uint64(), deckSize, handSize)
-		sort.Slice(hands, func(i, j int) bool {
-			return hands[i] < hands[j]
-		})
+	for i := uint64(0); i < hashMax; i++ {
+		ShuffleAndDealIntoHand(i, deckSize, aHand[:])
```

Comment:
FYI, using `i` rather than `rand.Uint64()` as the hash value made the distribution in the histogram much tighter.

Reply:
BTW, if we set hashMax to `uint64(fallingFactorial) * 16` then every hand gets dealt the same number of times.